drivers/storage/portworx/util/util.go

```diff
@@ -1403,12 +1407,34 @@ func CountStorageNodes(
 		return -1, fmt.Errorf("failed to enumerate nodes: %v", err)
 	}
 
+	// We need to find kubernetes nodes that can run storage pods only for DR clusters that use px versions lesser than 3.1.1
+	k8sNodesStoragePodCouldRun := make(map[string]bool)
+	if pxVersion.LessThan(MinimumPxVersionCurrentDomain) && isDRSetup {
```

Comment:
Should we move this after line 1455 and update the condition to ` isDRSetup && !useClusterDomain `?

This way we would have compared the px versions for all nodes. During a PX upgrade, until all PX nodes are not upgraded to 3.1.1, we should continue to calculate PDB value as before. 

`pxVersion` is the desired version from StorageCluster, it does not mean all the nodes are on that version.